src/olympia/shelves/serializers.py

```diff
@@ -43,9 +43,12 @@ def get_addons(self, obj):
             criteria = obj.criteria.strip('?')
             params = dict(parse.parse_qsl(criteria))
             request = self.context.get('request', None)
+            tmp = request.GET
             request.GET = request.GET.copy()
             request.GET.update(params)
-            return AddonSearchView(request=request).data
+            addons = AddonSearchView(request=request).data
+            request.GET = tmp
```

Comment:
if we had to do this in a few different places we could write it as a decorator to be used with a `with` statement (it would make the changes to request and revert them at the end) - not worth the extra code/complexity for one occurrence though.